modules/gatsby/src/gatsby-node/process-nodes-markdown.js

```diff
@@ -105,24 +124,39 @@ module.exports.processNewMarkdownNode = function processNewMarkdownNode({ node,
     createNodeField({ node, name: 'slug', value: relPath });
     node.frontmatter.path = relPath;
   }
-
-  docNodes[relPath] = node;
-}
+  if (tocNode) {
+    // this means toc node has been created. Any markdown file processed beyond this point wouldn't have its info
+    // in the toc.
+    // but we can inject it afterwards
+    const nodeToEdit = parseToc([tocNode], relPath);
+    if (nodeToEdit) {
+      nodeToEdit.childMarkdownRemark = {
```

Comment:
the regular toc node generation process adds the full content of each markdown node to the toc. we don't need as much. The app will only use the title and slug of the corresponding markdown node for each toc entry.

Reply:
Suggestion: When I feel the need to add a comment like that to the PR to guide the reader, I usually realize that I would have been better of just adding it to the code so that future maintainers also benefit...